src/Engine/Component/RigidBody.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <glm/glm.hpp>
+#include "SuperComponent.hpp"
+
+class btRigidBody;
+class PhysicsManager;
+
+namespace Physics {
+    class Trigger;
+}
+
+namespace Component {
+    /// %Component that allows interacting with other physics components.
+    class RigidBody : public SuperComponent {
+        friend class ::PhysicsManager;
+        friend class ::Physics::Trigger;
+
+        public:
+            /// Constructor.
+            RigidBody() = default;
+
+            /// Destructor.
+            ~RigidBody();
+
+            /// Save the component.
+            /**
+             * @return JSON value to be stored on disk.
+             */
+            Json::Value Save() const override;
+
+        private:
```

Comment:
Duplicate private modifier.
Also private members shouldn't have doxygen comments.

Reply:
Regarding modifier: It's valid C++ and is used here to neatly structure methods as opposed to variables. Any particular reason why this should be altered?

Regarding comments: Shouldn't as in verboten or don't need to?

Reply:
Doxygen per default does not show private members in the generated documentation (and the end-user shouldn't have to care about them), but if a comment is written it will. (I think there's a setting to always ignore but I don't remember if we have it on.) Of course documenting private members with regular comments is still a good thing.

Reply:
Cool. I'll get right to it.